packages/addon-dev/src/rollup-keep-assets.ts

```diff
@@ -11,22 +11,43 @@ export default function keepAssets({
   from: string;
   include: string[];
 }): Plugin {
+  let sourceDir = join(process.cwd(), from);
+
   return {
     name: 'copy-assets',
 
     // imports of assets should be left alone in the source code. This can cover
     // the case of .css as defined in the embroider v2 addon spec.
     async resolveId(source, importer, options) {
+      // We need to check if the `source` (what the importer is trying to import)
+      // matches our patterns, because we don't want to hit the file system if
+      // the pattern doesn't match.
+      //
+      // Without doing so can multiply the build time by 3x.
+      let isMatch = include.some((pattern) => minimatch(source, pattern));
+
+      // It's possible that `source` is a relative import instead of absolute,
+      // so we need to generate that absolute path (or at least relative to the importer)
+      // and then see if that matches our globs
+      if (importer && !isMatch) {
+        let absolute = join(dirname(importer), source);
```

Comment:
This stuff is definitely not OK. It has a bunch of broken edge cases. And if you fix all the edge cases, what you're going to end up with is `this.resolve()` anyway, the thing you're trying not to do.